openpype/lib/events.py

```diff
@@ -17,174 +16,83 @@ class MissingEventSystem(Exception):
-        else:
-            # Convert callback into references
-            #   - deleted functions won't cause crashes
-            if inspect.ismethod(func):
-                func_ref = WeakMethod(func)
-            else:
-                func_ref = weakref.ref(func)
-
-            # Get expected arguments from function spec
-            # - positional arguments are always preferred
-            expect_args = is_func_signature_supported(func, "fake")
-            expect_kwargs = is_func_signature_supported(func, event="fake")
-
-        self._func_ref = func_ref
+    # Get path to file and fallback to '<unknown path>' if fails
+    # NOTE This was added because of 'partial' functions which is handled,
+    #   but who knows what else can cause this to fail?
+    try:
+        path = os.path.abspath(inspect.getfile(func))
+    except TypeError:
+        pass
 
-        self._is_partial = is_partial
-        self._sub_func = sub_func
-        self._args = args
-        self._kwargs = kwargs
-        self._ref_is_valid = None
-        self._expect_args = expect_args
-        self._expect_kwargs = expect_kwargs
+    return name, path
 
-        self._name = None
-        self._path = None
 
-        self._fill_func_info(func)
+class weakref_partial:
+    """Partial function with weak reference to the wrapped function.
 
-    @property
-    def name(self):
-        """Get name of function.
+    Can be used as 'functools.partial' but it will store weak reference to
+        function. That means that the function must be stored in memory
+        somewhere. e.g. lambda functions are not supported as valid callback.
 
-        Returns:
-            str: Name of function.
-        """
-
-        return self._name
-
-    @property
-    def path(self):
-        """Get path to file where function is defined.
-
-        Returns:
-            str: Path to file.
-        """
+    Is useful for object methods. In that case the callback is
+        deregistered when object is destroyed.
 
-        return self._path
+    Warnings:
+        Values passed as *args and **kwargs are stored strongly in memory.
+            That may "keep alive" objects that should be already destroyed.
+            It is recommended to pass only immutable objects like 'str',
+            'bool', 'int' etc.
 
-    @property
-    def expect_args(self):
-        """
+    Args:
+        func (Callable): Function to wrap.
+        *args: Arguments passed to the wrapped function.
+        **kwargs: Keyword arguments passed to the wrapped function.
+    """
 
-        Returns:
-            Callback expects arguments.
-        """
+    def __init__(self, func, *args, **kwargs):
+        self._func_ref = _get_func_ref(func)
+        self._args = args
+        self._kwargs = kwargs
 
-        return self._expect_args
+    def __call__(self, *args, **kwargs):
+        func = self._func_ref()
+        if func is not None:
+            return func(*self._args, **self._kwargs)
```

Comment:
This is not a drop-in replacement to `functools.partial` - since `functools.partial` allows to pass in additional arguments and kwargs to the function.